src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchCaseSearchOperation.java

```diff
@@ -62,7 +63,7 @@ public ElasticsearchCaseSearchOperation(JestClient jestClient,
     public CaseSearchResult execute(CrossCaseTypeSearchRequest request) {
         MultiSearchResult result = search(request);
         if (result.isSucceeded()) {
-            return toCaseDetailsSearchResult(result);
+            return toCaseDetailsSearchResult(result, request);
```

Comment:
I'm afraid this class has reached the max size limit. We can't keep adding more code to it. 
Can you please introduce a new class `ElasticsearchCaseSearchResponseGenerator` and move all the response generation code in there